src/backend/catalog/storage_pending_deletes_redo.c

```diff
@@ -0,0 +1,324 @@
+/*-------------------------------------------------------------------------
+ *
+ * storage_pending_deletes_redo.c
+ *	  code to support processing of pending deletes (orphaned files) in WAL
+ *
+ * Copyright (c) 2025 Greengage Community
+ *
+ *	  src/backend/catalog/storage_pending_deletes_redo.c
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/clog.h"
+#include "access/transam.h"
+#include "catalog/storage_pending_deletes_redo.h"
+#include "miscadmin.h"
+#include "storage/md.h"
+#include "utils/elog.h"
+#include "utils/guc.h"
+#include "utils/hsearch.h"
+
+/*
+ * HTAB entry for pending deletes for the given xid.
+ */
+typedef struct PendingDeleteHtabNode
+{
+	TransactionId xid;
+	List	   *relnode_list;	/* list of RelFileNodePendingDelete */
+}	PendingDeleteHtabNode;
+
+/*
+ * Hash table for pending deletes relfilenodes for a given xid.
+ */
+static HTAB *pendingDeletesRedo = NULL;
+
+/*
+ * This function inserts XLOG_PENDING_DELETE record into WAL.
+ */
+void
+PdlXLogInsert()
+{
+	Size		size = 0;
+
+	if (IsBootstrapProcessingMode() || !gp_track_pending_delete)
```

Comment:
This condition is used in many places. What about a function like `is_tracking_enabled()`? https://github.com/arenadata/gpdb/pull/1460/files#diff-81bafef1e0ecb196ff356bd78b4915bec858f1956fc0acf80c8b01192af515e2R45